src/HDF5.jl

```diff
@@ -2153,23 +2200,12 @@ function get_chunk(dset::HDF5Dataset)
     end
     ret
 end
-set_chunk(p::HDF5Properties, dims...) = h5p_set_chunk(p.id, length(dims), hsize_t[reverse(dims)...])
-function get_userblock(p::HDF5Properties)
-    alen = Ref{hsize_t}()
-    h5p_get_userblock(p.id, alen)
-    alen[]
-end
-function get_fclose_degree(p::HDF5Properties)
-    out = Ref{Cint}()
-    h5p_get_fclose_degee(p.id, out)
-    out[]
-end
-function get_libver_bounds(p::HDF5Properties)
-    out1 = Ref{Cint}()
-    out2 = Ref{Cint}()
-    h5p_get_libver_bounds(p.id, out1, out2)
-    out1[], out2[]
-end
+
+get_alignment(p::HDF5Properties) = h5p_get_alignment(checkvalid(p).id)
```

Comment:
I guess previously these didn't `checkvalid`, I guess neither do the setters. I'm kind of wondering if we need a more thorough audit.

Reply:
There definitely needs to be an audit at some point. I added them here because `get_alignment` in my head is "higher level" than `h5p_get_alignment`, but that was a semi-arbitrary decision.

Reply:
I agree that if we add them they should be in this function. I was noting the behavioral difference, that this now checks them where as we didn't' before. The setters, also seem to be missing checks?

Reply:
I'm about to add a check to the `setindex!(::HDF5Properties, ...)` that's missing, so that'll handle most uses where properties are set through the keyword interface. It does look like, yes, there are still some stragglers like `set_chunk` that should probably gain a `checkvalid`, though.

Reply:
So since `getindex` does `checkvalid`, and the `get_*` functions also do the checking, this will end up double checking everything. 

Perhaps we should only check at the highest level function which would be `getindex`? Of course this would then mean independent calls to the getters would not check, which could be problematic.

Reply:
No — for all properties except `:chunk`, `_prop_get` calls the `h5p_get_*` functions, so the check won't be repeated. (`get_chunk` is used instead of `h5p_get_chunk` because the former is what handles reversing the order of dimensions.)

Reply:
Oh that's right